Neos.Media/Classes/Domain/Repository/AssetRepository.php

```diff
@@ -268,8 +268,9 @@ protected function addAssetCollectionToQueryConstraints(QueryInterface $query, A
             return;
         }
 
-        $constraints = $query->getConstraint();
-        $query->matching($query->logicalAnd([$constraints, $query->contains('assetCollections', $assetCollection)]));
+        $query->getQueryBuilder()->andWhere(
+            $query->contains('assetCollections', $assetCollection)
+        );
```

Comment:
The old code _looks_ as though it would just combine the existing constraints..
Ofc I trust you that this is not the case, but it would be great to have it covered by tests (in any case)

Reply:
I guess this has worked at some point and something in Doctrine has changed. Need to have a look into the tests, but I assume that just an functional/integration test can cover this.

Reply:
> something in Doctrine has changed

too bad o.O

> I assume that just an functional/integration test can cover this

Yes, I was thinking of one new method in
https://github.com/neos/neos-development-collection/blob/9.0/Neos.Media/Tests/Functional/Domain/Repository/AssetRepositoryTest.php

Reply:
Uh oh, that basically means the promises of the QueryInterface are broken? A test woudl be great but I am tagging @kdambekalns as well, I guess we might need to take more action in flow to warn people of this trap if it doesn'T work...

Reply:
Maybe this has never worked before, but it wasn't an issue. If I see that correctly ImageVariants have started to have asset collections with this PR of @kdambekalns, which we merged not that long ago.

https://github.com/neos/neos-development-collection/pull/3562

Reply:
I added the tests.

Reply:
It's not about `logicalAnd`. The constraints are empty, as the `andWhere` of the variant filter doesn't end up as a constraint:

https://github.com/neos/neos-development-collection/blob/e4da4f42ecb58c4b6805deb0d2920d4459a5227f/Neos.Media/Classes/Domain/Repository/AssetRepository.php#L282-L290

https://github.com/doctrine/orm/blob/2.17.x/lib/Doctrine/ORM/QueryBuilder.php#L1168-L1180